discord/component.go

```diff
@@ -822,6 +858,9 @@ type MentionableSelectComponent struct {
 	ValueLimits [2]int `json:"-"`
 	// Disabled disables the select if true.
 	Disabled bool `json:"disabled,omitempty"`
+	// SelectedMentions is the slice of discord.UserID's and discord.RoleID's 
+	// that are marked as selected by default
+	SelectedMentions []interface{} `json:"-"`
```

Comment:
If we're using a hidden field, then it's best to just have separate hidden `DefaultUsers` and `DefaultRoles` fields, then combine them in `MarshalJSON`.

Reply:
I agree with @MCausc78. What's wrong with using plain interfaces? It's already documented above the field for the users. 

I can add an example snippet in the comments.

Reply:
> What's wrong with using plain interfaces? It's already documented above the field for the users.

It's not type-safe.

The only way I would agree to moving forward with this solution is to make custom `SelectedUserID` and `SelectedRoleID` types and a `SelectableID` union type to combine the two. Arikawa tries its best to not have any ambiguous/error-prone `any` types in its code, so this shouldn't be an exception.

Reply:
@diamondburned How about this implementation?

```go
type DefaultMention struct {
	userId *UserID `json:"-"`
	roleId *RoleID `json:"-"`
}

// Creates new DefaultMention type with only UserID
func UserMention (userId UserID) DefaultMention {
	return DefaultMention{userId: &userId}
}

// Creates new DefaultMention type with only UserID
func RoleMention(roleId RoleID) DefaultMention {
	return DefaultMention{roleId: &roleId}
}
```

and when implementing,

```go
DefaultMentions: []discord.DefaultMention{ discord.UserMention(793688107077468171), discord.RoleMention(1299037008332849212) },
```

Reply:
I'm not sure about the naming, but this looks good. You can use `{UserID, RoleID}` for the types instead of `{*UserID, *RoleID}` then check for `IsValid()`.

Reply:
Commit done...! and for the namings, we can change it if the users are feeling discomfort with it.